### PR TITLE
Doc animation

### PR DESCRIPTION
**Pull request by tacaswell**
The plan (if I have time to get to it is to flesh out the FuncAnimation section with the same example implemented the other 3 ways, but I think that as-is it is still a major improvement.

### CODE REVIEW

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def adjusted_figsize(w, h, dpi, n):
 
 # A registry for available MovieWriter classes
 class MovieWriterRegistry(object):
+    '''Registry of of available writer classes by human readable name
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: double `of`

File: lib/matplotlib/animation.py [hunk 2]
```diff
@@ -109,6 +111,15 @@ def list(self):
         return list(self.avail.keys())
 
     def is_available(self, name):
+        '''If given writer is available
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: How about ```'''Check if given writer is available.'''``` ?

File: lib/matplotlib/animation.py [hunk 3]
```diff
@@ -109,6 +111,15 @@ def list(self):
...
+
+        Parameters
+        ----------
+        name : str
+
+        Returns
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Missing `-------` after "Returns"

File: lib/matplotlib/animation.py [hunk 4]
```diff
@@ -122,11 +133,36 @@ def __getitem__(self, name):
...
+
+    Attributes
+    ----------
+
+    frame_format : string
+        The format used in writing frame data, defaults to 'rgba'
+
+    fig : `~matplotlib.figure.Figure`
+        The figure to capture data from.
+        This must be provided by the sub-classes
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Missing period at the end of the sentence.

File: lib/matplotlib/animation.py [hunk 5]
```diff
@@ -122,11 +133,36 @@ def __getitem__(self, name):
...
+
+    Examples
+    --------
+
+    Fundamentally, what a MovieWriter does is provide is a way to grab
+    frames by calling grab_frame(). setup() is called to start the
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Should any of these functions (e.g. grab_frame()) be wrapped in ```` to make them look like code in the rendered docs?

File: lib/matplotlib/animation.py [hunk 6]
```diff
@@ -329,29 +366,37 @@ def isAvailable(cls):
...
+
+        '''
+
+        MovieWriter.__init__(self, *args, **kwargs)
+        self.frame_format = rcParams['animation.frame_format']
+
+    def setup(self, fig, outfile, dpi, frame_prefix='_tmp', clear_temp=True):
+        '''
+        Perform setup for writing the movie file.
+
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Extra blank line intentional?

File: lib/matplotlib/animation.py [hunk 7]
```diff
@@ -490,6 +541,11 @@ def output_args(self):
 # Combine FFMpeg options with pipe-based writing
 @writers.register('ffmpeg')
 class FFMpegWriter(MovieWriter, FFMpegBase):
+    '''Pipe based ffmpeg writer.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Technically, "pipe-based"...

File: lib/matplotlib/animation.py [hunk 8]
```diff
@@ -506,6 +562,12 @@ def _args(self):
 # Combine FFMpeg options with temp file-based writing
 @writers.register('ffmpeg_file')
 class FFMpegFileWriter(FileMovieWriter, FFMpegBase):
+    '''File based ffmpeg writer
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: As above, but "File-based"...

File: lib/matplotlib/animation.py [hunk 9]
```diff
@@ -520,20 +582,35 @@ def _args(self):
...
+
     exec_key = 'animation.avconv_path'
     args_key = 'animation.avconv_args'
 
 
 # Combine AVConv options with pipe-based writing
 @writers.register('avconv')
 class AVConvWriter(AVConvBase, FFMpegWriter):
-    pass
+    '''Pipe based avconv writer.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: See above.

File: lib/matplotlib/animation.py [hunk 10]
```diff
@@ -520,20 +582,35 @@ def _args(self):
...
+    Frames are streamed directly to avconv via a pipe and written in a single
+    pass.
+    '''
 
 
 # Combine AVConv options with file-based writing
 @writers.register('avconv_file')
 class AVConvFileWriter(AVConvBase, FFMpegFileWriter):
-    pass
+    '''File based avconv writer
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: See above.

File: lib/matplotlib/animation.py [hunk 11]
```diff
@@ -667,6 +750,12 @@ def isAvailable(cls):
 # former.
 @writers.register('imagemagick')
 class ImageMagickWriter(ImageMagickBase, MovieWriter):
+    '''Pipe based animated gif
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: As above.

File: lib/matplotlib/animation.py [hunk 12]
```diff
@@ -681,6 +770,13 @@ def _args(self):
 # former.
 @writers.register('imagemagick_file')
 class ImageMagickFileWriter(ImageMagickBase, FileMovieWriter):
+    '''File based animated gif writer
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: As above.

File: lib/matplotlib/animation.py [hunk 13]
```diff
@@ -758,55 +869,72 @@ def _stop(self, *args):
...
+
+        dpi : number, optional
+           controls the dots per inch for the movie frames. This
+           combined with the figure's size in inches controls the size of
+           the movie.  If None, defaults to ``rcparam['savefig.dpi']``
+
+        codec : str, optional
+           the video codec to be used. Not all codecs are supported by
+           a given :class:`MovieWriter`. If `None`,
+           default to  ``rcParams['animation.codec']``
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Extra space after "to"

File: lib/matplotlib/animation.py [hunk 14]
```diff
@@ -758,55 +869,72 @@ def _stop(self, *args):
...
+        savefig_kwargs : dict, optional
+           is a dictionary containing keyword arguments to be passed
+           on to the 'savefig' command which is called repeatedly to
+           save the individual frames.
+
+        Notes
+        -----
+        fps, codec, bitrate, extra_args, metadata are used are used to
+        construct a :class:`MovieWriter` instance and can only be
+        passed if `writer` is a string.  If they are pass as
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: "pass" -> "passed"

File: lib/matplotlib/animation.py [hunk 15]
```diff
@@ -1081,15 +1209,31 @@ def _repr_html_(self):
...
+    A new frame is drawn every *interval* milliseconds.
+
+    Parameters
+    ----------
+    fig : matplotlib.figure.Figure
+       The figure object that is used to get draw, resize, and any
+       other needed events.
+
+    interval : number, optional
+       Delay between frames.  Defaults to 200
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Period after 200?

File: lib/matplotlib/animation.py [hunk 16]
```diff
@@ -1081,15 +1209,31 @@ def _repr_html_(self):
...
+
+    repeat_delay : number, optional
+        If the animation in repeated, adds a delay in milliseconds
+        before repeating the animation.  Defaults to None
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: "None" -> "`None`."

File: lib/matplotlib/animation.py [hunk 17]
```diff
@@ -1081,15 +1209,31 @@ def _repr_html_(self):
...
 
-    *repeat* controls whether the animation should repeat when the sequence
-    of frames is completed.
+    repeat: bool, optional
+        controls whether the animation should repeat when the sequence
+        of frames is completed.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Add "Defaults to `True`."

File: lib/matplotlib/animation.py [hunk 18]
```diff
@@ -1147,13 +1291,38 @@ def _loop_delay(self, *args):
...
+       The figure object that is used to get draw, resize, and any
+       other needed events.
+
+    artists : list
+        with each list entry a collection of artists that
+        represent what needs to be enabled on each frame. These will
+        be disabled for other frames.
+
+    interval : number, optional
+       Delay between frames.  Defaults to 200
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Period after 200?

File: lib/matplotlib/animation.py [hunk 19]
```diff
@@ -1147,13 +1291,38 @@ def _loop_delay(self, *args):
...
+
+    repeat_delay : number, optional
+        If the animation in repeated, adds a delay in milliseconds
+        before repeating the animation.  Defaults to None
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: "None" -> "`None`."

File: lib/matplotlib/animation.py [hunk 20]
```diff
@@ -1147,13 +1291,38 @@ def _loop_delay(self, *args):
...
+    fig : matplotlib.figure.Figure
+       The figure object that is used to get draw, resize, and any
+       other needed events.
+
+    artists : list
+        with each list entry a collection of artists that
+        represent what needs to be enabled on each frame. These will
+        be disabled for other frames.
+
+    interval : number, optional
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Broader question: Do we need to repeat the documentation of attributes from the base class?

File: lib/matplotlib/animation.py [hunk 21]
```diff
@@ -1147,13 +1291,38 @@ def _loop_delay(self, *args):
...
+       Delay between frames.  Defaults to 200
+
+    repeat_delay : number, optional
+        If the animation in repeated, adds a delay in milliseconds
+        before repeating the animation.  Defaults to None
+
+    repeat: bool, optional
+        controls whether the animation should repeat when the sequence
+        of frames is completed.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Add "Defaults to `True`."

File: lib/matplotlib/animation.py [hunk 22]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+
+
+    Parameters
+    ----------
+    fig : matplotlib.figure.Figure
+       The figure object that is used to get draw, resize, and any
+       other needed events.
+
+    func : callable
+
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Extra blank line.

File: lib/matplotlib/animation.py [hunk 23]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+       The function to call at each frame.  The first argument will
+       be the next value in ``frames``.   Any additional positional
+       arguments can be supplied via the ``fargs`` parameter.
+
+       The required signature is ::
+
+          def func(fr: object, *fargs) -> iterable_of_artists:
+
+
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Not Sure if this is an extra blank line or one needed to surround the above code block...

File: lib/matplotlib/animation.py [hunk 24]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+        If an integer, equivalent to passing ``range(frames)``
+
+        If a generator function, then must have the signature ::
+
+           def gen_function():
+
+        If `None`, then equivalent to passing ``itertools.count``
+
+    init_func : callable, optional
+
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Extra blank line

File: lib/matplotlib/animation.py [hunk 25]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+          def init_func() -> iterable_of_artists:
+
+    fargs : tuple or None, optional
+       Additional arguments to pass to each call to ``func``
+
+    save_count : int, optional
+       The number of values from `frames` to cache.
+
+    interval : number, optional
+       Delay between frames.  Defaults to 200
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Period after 200?

File: lib/matplotlib/animation.py [hunk 26]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+    repeat_delay : number, optional
+       If the animation in repeated, adds a delay in milliseconds
+       before repeating the animation.  Defaults to None
 
-    *init_func* is a function used to draw a clear frame. If not given, the
-    results of drawing from the first item in the frames sequence will be
-    used. This function will be called once before the first frame.
+    repeat: bool, optional
+       controls whether the animation should repeat when the sequence
+       of frames is completed.
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Add "Defaults to `True`."

File: lib/matplotlib/animation.py [hunk 27]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+    save_count : int, optional
+       The number of values from `frames` to cache.
+
+    interval : number, optional
+       Delay between frames.  Defaults to 200
 
-    *frames* can be a generator, an iterable, or a number of frames.
+    repeat_delay : number, optional
+       If the animation in repeated, adds a delay in milliseconds
+       before repeating the animation.  Defaults to None
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: "None" -> "`None`."

File: lib/matplotlib/animation.py (hunk 20 above)
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Ok, you've convinced me.

File: lib/matplotlib/animation.py [hunk 28]
```diff
@@ -122,11 +133,36 @@ def __getitem__(self, name):
...
+        The format used in writing frame data, defaults to 'rgba'
+
+    fig : `~matplotlib.figure.Figure`
+        The figure to capture data from.
+        This must be provided by the sub-classes
+
+    Examples
+    --------
+
+    Fundamentally, what a MovieWriter does is provide is a way to grab
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Remove second 'is'.

File: lib/matplotlib/animation.py [hunk 29]
```diff
@@ -136,15 +172,13 @@ class MovieWriter(object):
...
     '''
 
     def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
                  metadata=None):
         '''
-        Construct a new MovieWriter object.
-
+        Parameters
+        ----------
         fps: int
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Maybe change these to add the space before the colon that's needed while you're here, as done below.

File: lib/matplotlib/animation.py [hunk 30]
```diff
@@ -463,6 +508,12 @@ def cleanup(self):
 # Base class of ffmpeg information. Has the config keys and the common set
 # of arguments that controls the *output* side of things.
 class FFMpegBase(object):
+    '''Mixin class for FFMpeg output
+
+    To be useful this must be multiply-inherited from with a
+    `MoveWriterBase` sub-class.
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Move -> Movie

File: lib/matplotlib/animation.py [hunk 31]
```diff
@@ -520,20 +582,35 @@ def _args(self):
 # Base class of avconv information.  AVConv has identical arguments to
 # FFMpeg
 class AVConvBase(FFMpegBase):
+    '''Mixin class for avconv output
+
+    To be useful this must be multiply-inherited from with a
+    `MoveWriterBase` sub-class.
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Move -> Movie

File: lib/matplotlib/animation.py [hunk 32]
```diff
@@ -616,6 +693,12 @@ def _args(self):
 
 # Base class for animated GIFs with convert utility
 class ImageMagickBase(object):
+    '''Mixin class for ImageMagick output
+
+    To be useful this must be multiply-inherited from with a
+    `MoveWriterBase` sub-class.
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Move -> Movie

File: lib/matplotlib/animation.py [hunk 33]
```diff
@@ -758,55 +869,72 @@ def _stop(self, *args):
...
-        *metadata* is a dictionary of keys and values for metadata to include
-        in the output file. Some keys that may be of use include:
-        title, artist, genre, subject, copyright, srcform, comment.
+        '''Saves a movie file by drawing every frame.
+
+        Parameters
+        ----------
+
+        filename : str
+            the output filename, e.g., :file:`mymovie.mp4`
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: It's odd that some of these start with a lower-case letter, but then are followed by another capitalized sentence.

File: lib/matplotlib/animation.py [hunk 34]
```diff
@@ -758,55 +869,72 @@ def _stop(self, *args):
...
+           combined with the figure's size in inches controls the size of
+           the movie.  If None, defaults to ``rcparam['savefig.dpi']``
+
+        codec : str, optional
+           the video codec to be used. Not all codecs are supported by
+           a given :class:`MovieWriter`. If `None`,
+           default to  ``rcParams['animation.codec']``
+
+        bitrate : number, optional
+           specifies the amount of bits used per second in the
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: amount -> number

File: lib/matplotlib/animation.py [hunk 35]
```diff
@@ -758,55 +869,72 @@ def _stop(self, *args):
...
+           animations.
+
+        savefig_kwargs : dict, optional
+           is a dictionary containing keyword arguments to be passed
+           on to the 'savefig' command which is called repeatedly to
+           save the individual frames.
+
+        Notes
+        -----
+        fps, codec, bitrate, extra_args, metadata are used are used to
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Double "are used"

File: lib/matplotlib/animation.py [hunk 36]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
+        Source of data to pass ``func`` and each frame of the animation
+
+        If an iterable, then simply use the values provided.  If the
+        iterable has a length, it will override the ``save_count`` kwarg.
+
+        If an integer, equivalent to passing ``range(frames)``
+
+        If a generator function, then must have the signature ::
+
+           def gen_function():
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Should be `-> int` or `-> float` or anything?

File: lib/matplotlib/animation.py (hunk 25 above)
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Units?

File: lib/matplotlib/animation.py [hunk 37]
```diff
@@ -1202,25 +1371,75 @@ def _draw_frame(self, artists):
...
 
-    *frames* can be a generator, an iterable, or a number of frames.
+    repeat_delay : number, optional
+       If the animation in repeated, adds a delay in milliseconds
+       before repeating the animation.  Defaults to None
 
-    *init_func* is a function used to draw a clear frame. If not given, the
-    results of drawing from the first item in the frames sequence will be
-    used. This function will be called once before the first frame.
+    repeat: bool, optional
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Space before colon.

File: lib/matplotlib/animation.py (hunk 15 above)
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Units?

File: lib/matplotlib/animation.py (hunk 18 above)
Comment (QuLogic) [inline lib/matplotlib/animation.py]: Units?

**Comment by tacaswell** [comment]
I think I have addressed everything and re-worked the section on writers.

File: doc/api/animation_api.rst [hunk 38]
```diff
@@ -1,12 +1,333 @@
...
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   FFMpegFileWriter
+   ImageMagickWriter
+   AVConvFileWriter
+
+
+Fundamentally, a `MovieWriter` does is provide is a way to grab
```
Comment (QuLogic) [inline doc/api/animation_api.rst]: The 'what' got removed, so now the sentence is a bit off.

File: lib/matplotlib/animation.py [hunk 39]
```diff
@@ -463,6 +483,12 @@ def cleanup(self):
 # Base class of ffmpeg information. Has the config keys and the common set
 # of arguments that controls the *output* side of things.
 class FFMpegBase(object):
+    '''Mixin class for FFMpeg output
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: It's a bit pedantic, but the first line on each of these newly documented classes sometimes does and sometimes doesn't end with a period.

File: lib/matplotlib/animation.py [hunk 40]
```diff
@@ -1147,13 +1266,38 @@ def _loop_delay(self, *args):
...
+       The figure object that is used to get draw, resize, and any
+       other needed events.
+
+    artists : list
+        With each list entry a collection of artists that
+        represent what needs to be enabled on each frame. These will
+        be disabled for other frames.
+
+    interval : number, optional
+       Delay between frames in miliseconds.  Defaults to 200.
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: miliseconds -> milliseconds

File: lib/matplotlib/animation.py [hunk 41]
```diff
@@ -1081,15 +1184,31 @@ def _repr_html_(self):
...
+       other needed events.
+
+    interval : number, optional
+       Delay between frames in milliseconds.  Defaults to 200.
 
-    *repeat* controls whether the animation should repeat when the sequence
-    of frames is completed.
+    repeat_delay : number, optional
+        If the animation in repeated, adds a delay in milliseconds
+        before repeating the animation.  Defaults to None
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: This one's missing the backticks on `None`.

File: lib/matplotlib/animation.py [hunk 42]
```diff
@@ -1202,25 +1346,76 @@ def _draw_frame(self, artists):
...
+    repeat_delay : number, optional
+       If the animation in repeated, adds a delay in milliseconds
+       before repeating the animation.  Defaults to `None`.
 
-    *init_func* is a function used to draw a clear frame. If not given, the
-    results of drawing from the first item in the frames sequence will be
-    used. This function will be called once before the first frame.
+    repeat : bool, optional
+       Controls whether the animation should repeat when the sequence
+       of frames is completed.  Defaults to `True`
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: This one's missing the period.

File: doc/api/animation_api.rst (hunk 38 above)
Comment (QuLogic) [inline doc/api/animation_api.rst]: And the second 'is' is still there.

File: doc/api/animation_api.rst [hunk 43]
```diff
@@ -1,12 +1,333 @@
...
+sequential frames from the same underlying `~matplotlib.figure.Figure`
+object.  The base class `MovieWriter` implements 3 methods and a
+context manager.  The only difference between the pipe-based and
+file-based writers in the arguments to their respective ``setup``
```
Comment (QuLogic) [inline doc/api/animation_api.rst]: in ->is in

File: doc/api/animation_api.rst [hunk 44]
```diff
@@ -1,12 +1,333 @@
...
+   MovieWriter.saving
+
+
+The ``setup()`` method is used to prepare the writer (possibly opening
+a pipe), successive calls to ``grab_frame()`` capture a single frame
+at a time and ``finish()`` finalizes the movie and writes the output
+file to disk.  For example ::
+
+   moviewriter = MovieWriter(...)
+   moveiewriter.setup(fig=fig, 'my_movie.ext', dpi=100)
```
Comment (QuLogic) [inline doc/api/animation_api.rst]: moveiewriter -> moviewriter

File: lib/matplotlib/animation.py [hunk 45]
```diff
@@ -758,55 +844,72 @@ def _stop(self, *args):
...
+           combined with the figure's size in inches controls the size of
+           the movie.  If None, defaults to ``rcparam['savefig.dpi']``
+
+        codec : str, optional
+           The video codec to be used. Not all codecs are supported by
+           a given :class:`MovieWriter`. If `None`,
+           default to ``rcParams['animation.codec']``
+
+        bitrate : number, optional
+           Specifies the amount of bits used per second in the
```
Comment (QuLogic) [inline lib/matplotlib/animation.py]: amount -> number

File: lib/matplotlib/animation.py [hunk 46]
```diff
@@ -1199,7 +1199,7 @@ class TimedAnimation(Animation):
 
     repeat_delay : number, optional
         If the animation in repeated, adds a delay in milliseconds
-        before repeating the animation.  Defaults to None
+        before repeating the animation.  Defaults to `None`
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Pedantic, but no period at the end.

File: lib/matplotlib/animation.py [hunk 47]
```diff
@@ -141,42 +141,19 @@ class MovieWriter(object):
...
-    The use of the context manager ensures that setup and cleanup are
-    performed as necessary.
+        This must be provided by the sub-classes.
 
     '''
 
     def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
                  metadata=None):
-        '''
+        '''MovieWriter
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: No period at the end--though not really a sentence, either. 😁

File: lib/matplotlib/animation.py [hunk 48]
```diff
@@ -1304,24 +1279,24 @@ class ArtistAnimation(TimedAnimation):
        other needed events.
 
     artists : list
-        with each list entry a collection of artists that
+        With each list entry a collection of artists that
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Maybe instead just: `Each list entry is a collection...`

File: lib/matplotlib/animation.py (hunk 47 above)
Comment (tacaswell) [inline lib/matplotlib/animation.py]: I intentionally decided against a `.` here (due to it not being a sentence).

Comment (dopplershift) [inline lib/matplotlib/animation.py]: Fair enough.

**Pull request merged**